### PR TITLE
Refactor: Remove old `aimag` runtime implementations

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Towards: https://github.com/lfortran/lfortran/issues/3034

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
```fortran
program expr2
implicit none

complex :: x
print *, aimag(x)

end program
```
I get the following error for the above example in WASM backend
Error:
```
code generation error: FunctionCall: Function _lfortran_caimag not found


Note: Please report unclear or confusing messages as bugs at
https://github.com/lfortran/lfortran/issues
```
Does WASM backend support BindC function? If it doesn't we need to disable the following integration_tests
```
Errors while running CTest
	132 - complex_03 (Failed)
	133 - complex_04 (Failed)
	206 - complex_11 (Failed)
	207 - complex_12 (Failed)
	208 - complex_13 (Failed)
	209 - complex_14 (Failed)
```

**Comment by certik** [comment]
WASM doesn't call into C, so you can't call any C function. For now.

**Comment by certik** [comment]
If WASM worked before, then it should work now. Maybe before it worked without calling into C?

**Comment by Thirumalai-Shaktivel** [comment]
Nope, I just checked it with the main branch.
It generates the following code and call into C.
```fortran
! Fortran code after applying the pass: insert_deallocate
module lfortran_intrinsic_math
implicit none
interface aimag
    module procedure caimag
end interface aimag

contains

elemental real(4) function caimag(x) result(r)
    complex(4), intent(in) :: x
    interface
        pure real(4) function c_caimag(x) bind(c, name = "_lfortran_caimag")
            complex(4), intent(in), value :: x
        end function c_caimag

    end interface
    r = c_caimag(x)
end function caimag

end module lfortran_intrinsic_math
program expr2
use lfortran_intrinsic_math, only: aimag
use lfortran_intrinsic_math, only: caimag
implicit none
complex(4), save :: x = (1.000000, 3.000000)
print *, aimag(x)
end program expr2
```
There seems to be some bug, I will debug and report back.

**Comment by Thirumalai-Shaktivel** [comment]
Fixed here: https://github.com/lfortran/lfortran/pull/3235/commits/1d6be5d2b74655348016d03e59d74022805c5f6f

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -1368,6 +1370,52 @@ create_trig(Sinh, sinh, sinh)
...
+
+    static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
+            diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args == 1,
+            "Call to `Aimag` must have exactly one argument",
+            x.base.base.loc, diagnostics);
+        ASRUtils::require_impl(is_complex(*expr_type(x.m_args[0])),
+            "Argument to `Aimag` must be of complex type",
+            x.base.base.loc, diagnostics);
+    }
```
Comment (certik) [inline src/libasr/pass/intrinsic_function_registry.h:1383]: Can you use the new Python script to generate this function automatically? Let's start using it.

**Comment by certik** [comment]
Very good, just left one minor comment.

**Comment by Thirumalai-Shaktivel** [comment]
Ready!

File: src/libasr/pass/intrinsic_function_registry.h (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply src/libasr/pass/intrinsic_function_registry.h]: Done!

**Review by certik — APPROVED** [review]

**Pull request merged**